### ISSUE TITLE
Amesos2 + Superludist 5.3.0: compilation issues with float

### ISSUE THREAD

**Issue by davydden** [trilinos/Trilinos#3566]
PackageName:  Amesos2
@trilinos/amesos2

I see a lot of compilation issues with Amesos2 when enabling
```
-DTpetra_INST_COMPLEX_FLOAT=ON
-DTeuchos_ENABLE_FLOAT=ON
```

## Expectations

Trilinos should build with parameters outline above.

## Current Behavior
Compiler errors:

```
/spack/stage/trilinos-12.12.1-i5lgamlpcrzplkscjaus6oe3wfv7ps6j/Trilinos-trilinos-release-12-12-1/packages/amesos2/src/Amesos2_Superludist_decl.hpp:115:30: error: no type named 'type' in 'Amesos2::TypeMap<Amesos2::Superludist, float>'
  typedef typename type_map::type                                  slu_type;
          ~~~~~~~~~~~~~~~~~~~^~~~
/spack/stage/trilinos-12.12.1-i5lgamlpcrzplkscjaus6oe3wfv7ps6j/Trilinos-trilinos-release-12-12-1/packages/amesos2/src/Amesos2_Superludist.cpp:62:30: note: in instantiation of template class 'Amesos2::Superludist<Tpetra::CrsMatrix<float, int, int, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::Serial, Kokkos::HostSpace>, false>, Tpetra::MultiVector<float, int, int, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::Serial, Kokkos::HostSpace>, false> >' requested here
  AMESOS2_SOLVER_TPETRA_INST(Superludist,float,int,int);
                             ^
In file included from /spack/stage/trilinos-12.12.1-i5lgamlpcrzplkscjaus6oe3wfv7ps6j/Trilinos-trilinos-release-12-12-1/packages/amesos2/src/Amesos2_Superludist.cpp:47:
/spack/stage/trilinos-12.12.1-i5lgamlpcrzplkscjaus6oe3wfv7ps6j/Trilinos-trilinos-release-12-12-1/packages/amesos2/src/Amesos2_Superludist_decl.hpp:116:30: error: no type named 'magnitude_type' in 'Amesos2::TypeMap<Amesos2::Superludist, float>'
  typedef typename type_map::magnitude_type                  magnitude_type;
          ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~
/spack/stage/trilinos-12.12.1-i5lgamlpcrzplkscjaus6oe3wfv7ps6j/Trilinos-trilinos-release-12-12-1/packages/amesos2/src/Amesos2_Superludist_decl.hpp:275:24: error: no type named 'LUstruct_t' in 'Amesos2::TypeMap<Amesos2::Superludist, float>'
    typename type_map::LUstruct_t LU; ///< storage for L and U factors
    ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
```

and more the same.

## Steps to Reproduce

configure `Trilinos` 12.12.1 with (among others)
```
-DTrilinos_ENABLE_EXPLICIT_INSTANTIATION:BOOL=ON
-DTpetra_INST_DOUBLE:BOOL=ON
-DTpetra_INST_INT_UNSIGNED_LONG=ON
-DTpetra_INST_INT_LONG:BOOL=ON
-DTpetra_INST_COMPLEX_DOUBLE=ON
-DTpetra_INST_COMPLEX_FLOAT=ON
-DTpetra_INST_FLOAT=ON
-DTeuchos_ENABLE_COMPLEX=ON
-DTeuchos_ENABLE_FLOAT=ON
```
and `superlu_dist` 5.3.0

## Your Environment
- macOS Mojave
- Apple Clang 10.0.0
- gfortran 8.2.0

p.s. full configure line
```
'cmake' '/Users/davydden/spack/var/spack/stage/trilinos-12.12.1-i5lgamlpcrzplkscjaus6oe3wfv7ps6j/Trilinos-trilinos-release-12-12-1' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:PATH=/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/trilinos-12.12.1-i5lgamlpcrzplkscjaus6oe3wfv7ps6j' '-DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_FIND_FRAMEWORK:STRING=LAST' '-DCMAKE_FIND_APPBUNDLE:STRING=LAST' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=FALSE' '-DCMAKE_INSTALL_RPATH:STRING=/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/trilinos-12.12.1-i5lgamlpcrzplkscjaus6oe3wfv7ps6j/lib;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/trilinos-12.12.1-i5lgamlpcrzplkscjaus6oe3wfv7ps6j/lib64;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/boost-1.68.0-t5bwyiollauogff4wxhvo52lm3sv7r2u/lib;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/bzip2-1.0.6-syohzw57v2jfag5du2x4bowziw3m5p67/lib;/usr/lib;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/zlib-1.2.11-uw6agnye4za7cpgyyaaejrzorhbhqno5/lib;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/glm-0.9.7.1-wkxhidyu42nywa5cqut2lhgiorvxp2t3/lib;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/hdf5-1.10.3-k3u6os3dvgcpi5vyfqfemq2y2h24qzna/lib;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/openmpi-3.1.2-g4aftkkwh42mabwyzqqdl6ahfqezutia/lib;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/hwloc-1.11.9-irsuwjwkkbbnffiov6majcobdhisitqv/lib;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/libxml2-2.9.8-5ansovhx566pvwyd6gldjmlplfujn4kd/lib;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/xz-5.2.4-i47twa3ynjmijdjvt4nag47wtg65m7jc/lib;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/hypre-2.14.0-3g745hjnw2mpjch7q3ck6njv3jvntfcs/lib;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/openblas-0.3.3-as2bzwl5gxnhzztryywyo5svliil55eg/lib;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/matio-1.5.9-uzbwa4ftmz7o2o5i2gtbyrom45rq6daw/lib;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/metis-5.1.0-6rrzripqqoszckfuh7yi3nfosysopqsh/lib;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/mumps-5.1.1-g3i6mskheyphqju3vs2cjpjtzjz5bq56/lib;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/netlib-scalapack-2.0.2-zxvnbswlaqg7hqfcifzvm4epd4wpjic6/lib;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/netcdf-4.6.1-eaaikpgal4dn4436523uhovdum2dlvql/lib;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/parmetis-4.0.3-othzzjoecn5ggse5phajddhwkloe33w3/lib;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/suite-sparse-5.3.0-3xnpg5pctoewgswrn5mlyl76qsyotz5y/lib;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/superlu-dist-5.3.0-26jrwc63ytki3dooo75r2pnltpi2w7x6/lib' '-DCMAKE_PREFIX_PATH:STRING=/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/hypre-2.14.0-3g745hjnw2mpjch7q3ck6njv3jvntfcs;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/cmake-3.12.2-xdk2owjni2wvf5w6jd2w3higkadgedk3;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/glm-0.9.7.1-wkxhidyu42nywa5cqut2lhgiorvxp2t3;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/superlu-dist-5.3.0-26jrwc63ytki3dooo75r2pnltpi2w7x6;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/metis-5.1.0-6rrzripqqoszckfuh7yi3nfosysopqsh;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/netlib-scalapack-2.0.2-zxvnbswlaqg7hqfcifzvm4epd4wpjic6;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/suite-sparse-5.3.0-3xnpg5pctoewgswrn5mlyl76qsyotz5y;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/openmpi-3.1.2-g4aftkkwh42mabwyzqqdl6ahfqezutia;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/netcdf-4.6.1-eaaikpgal4dn4436523uhovdum2dlvql;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/mumps-5.1.1-g3i6mskheyphqju3vs2cjpjtzjz5bq56;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/matio-1.5.9-uzbwa4ftmz7o2o5i2gtbyrom45rq6daw;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/boost-1.68.0-t5bwyiollauogff4wxhvo52lm3sv7r2u;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/parmetis-4.0.3-othzzjoecn5ggse5phajddhwkloe33w3;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/openblas-0.3.3-as2bzwl5gxnhzztryywyo5svliil55eg;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/hdf5-1.10.3-k3u6os3dvgcpi5vyfqfemq2y2h24qzna' '-DTrilinos_VERBOSE_CONFIGURE:BOOL=OFF' '-DTrilinos_ENABLE_TESTS:BOOL=OFF' '-DTrilinos_ENABLE_EXAMPLES:BOOL=OFF' '-DTrilinos_ENABLE_CXX11:BOOL=ON' '-DBUILD_SHARED_LIBS:BOOL=ON' '-DTPL_ENABLE_MPI:BOOL=ON' '-DCMAKE_C_COMPILER=/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/openmpi-3.1.2-g4aftkkwh42mabwyzqqdl6ahfqezutia/bin/mpicc' '-DCMAKE_CXX_COMPILER=/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/openmpi-3.1.2-g4aftkkwh42mabwyzqqdl6ahfqezutia/bin/mpic++' '-DCMAKE_Fortran_COMPILER=/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/openmpi-3.1.2-g4aftkkwh42mabwyzqqdl6ahfqezutia/bin/mpif90' '-DMPI_BASE_DIR:PATH=/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/openmpi-3.1.2-g4aftkkwh42mabwyzqqdl6ahfqezutia' '-DTrilinos_ENABLE_ALL_OPTIONAL_PACKAGES:BOOL=OFF' '-DTrilinos_ENABLE_Amesos:BOOL=ON' '-DTrilinos_ENABLE_Amesos2:BOOL=ON' '-DTrilinos_ENABLE_Anasazi:BOOL=ON' '-DTrilinos_ENABLE_AztecOO:BOOL=ON' '-DTrilinos_ENABLE_Belos:BOOL=ON' '-DTrilinos_ENABLE_Epetra:BOOL=ON' '-DTrilinos_ENABLE_EpetraExt:BOOL=ON' '-DTrilinos_ENABLE_Ifpack:BOOL=ON' '-DTrilinos_ENABLE_Ifpack2:BOOL=ON' '-DTrilinos_ENABLE_Intrepid=OFF' '-DTrilinos_ENABLE_Intrepid2=OFF' '-DTrilinos_ENABLE_Isorropia=OFF' '-DTrilinos_ENABLE_Kokkos:BOOL=ON' '-DTrilinos_ENABLE_MiniTensor=OFF' '-DTrilinos_ENABLE_ML:BOOL=ON' '-DTrilinos_ENABLE_MueLu:BOOL=ON' '-DTrilinos_ENABLE_NOX:BOOL=OFF' '-DTrilinos_ENABLE_Piro:BOOL=OFF' '-DTrilinos_ENABLE_Phalanx=OFF' '-DTrilinos_ENABLE_PyTrilinos:BOOL=OFF' '-DTrilinos_ENABLE_ROL:BOOL=ON' '-DTrilinos_ENABLE_Rythmos=OFF' '-DTrilinos_ENABLE_Sacado:BOOL=ON' '-DTrilinos_ENABLE_Shards=OFF' '-DTrilinos_ENABLE_Teko=OFF' '-DTrilinos_ENABLE_Tempus=OFF' '-DTrilinos_ENABLE_Teuchos:BOOL=ON' '-DTrilinos_ENABLE_Tpetra:BOOL=ON' '-DTrilinos_ENABLE_Zoltan:BOOL=ON' '-DTrilinos_ENABLE_Zoltan2:BOOL=ON' '-DTrilinos_ENABLE_SEACAS:BOOL=ON' '-DTrilinos_ENABLE_SEACASExodus:BOOL=ON' '-DTrilinos_ENABLE_SEACASEpu:BOOL=ON' '-DTrilinos_ENABLE_SEACASExodiff:BOOL=ON' '-DTrilinos_ENABLE_SEACASNemspread:BOOL=ON' '-DTrilinos_ENABLE_SEACASNemslice:BOOL=ON' '-DTrilinos_ENABLE_SEACASIoss:BOOL=ON' '-DTPL_ENABLE_BLAS=ON' '-DBLAS_LIBRARY_NAMES=openblas' '-DBLAS_LIBRARY_DIRS=/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/openblas-0.3.3-as2bzwl5gxnhzztryywyo5svliil55eg/lib' '-DTPL_ENABLE_LAPACK=ON' '-DLAPACK_LIBRARY_NAMES=openblas' '-DLAPACK_LIBRARY_DIRS=/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/openblas-0.3.3-as2bzwl5gxnhzztryywyo5svliil55eg/lib' '-DTPL_ENABLE_Netcdf:BOOL=ON' '-DNetCDF_ROOT:PATH=/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/netcdf-4.6.1-eaaikpgal4dn4436523uhovdum2dlvql' '-DTPL_ENABLE_X11:BOOL=OFF' '-DTrilinos_ENABLE_Gtest:BOOL=ON' '-DTPL_ENABLE_HYPRE:BOOL=ON' '-DHYPRE_INCLUDE_DIRS:PATH=/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/hypre-2.14.0-3g745hjnw2mpjch7q3ck6njv3jvntfcs/include' '-DHYPRE_LIBRARY_DIRS:PATH=/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/hypre-2.14.0-3g745hjnw2mpjch7q3ck6njv3jvntfcs/lib' '-DTPL_ENABLE_Boost:BOOL=ON' '-DBoost_INCLUDE_DIRS:PATH=/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/boost-1.68.0-t5bwyiollauogff4wxhvo52lm3sv7r2u/include' '-DBoost_LIBRARY_DIRS:PATH=/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/boost-1.68.0-t5bwyiollauogff4wxhvo52lm3sv7r2u/lib' '-DTPL_ENABLE_HDF5:BOOL=ON' '-DHDF5_INCLUDE_DIRS:PATH=/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/hdf5-1.10.3-k3u6os3dvgcpi5vyfqfemq2y2h24qzna/include' '-DHDF5_LIBRARY_DIRS:PATH=/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/hdf5-1.10.3-k3u6os3dvgcpi5vyfqfemq2y2h24qzna/lib' '-DTPL_ENABLE_Cholmod:BOOL=OFF' '-DTPL_ENABLE_UMFPACK:BOOL=ON' '-DUMFPACK_LIBRARY_DIRS:PATH=/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/suite-sparse-5.3.0-3xnpg5pctoewgswrn5mlyl76qsyotz5y/lib' '-DUMFPACK_INCLUDE_DIRS:PATH=/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/suite-sparse-5.3.0-3xnpg5pctoewgswrn5mlyl76qsyotz5y/include' '-DUMFPACK_LIBRARY_NAMES=umfpack;amd;colamd;cholmod;suitesparseconfig' '-DTPL_ENABLE_METIS:BOOL=ON' '-DMETIS_LIBRARY_DIRS=/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/metis-5.1.0-6rrzripqqoszckfuh7yi3nfosysopqsh/lib' '-DMETIS_LIBRARY_NAMES=metis' '-DTPL_METIS_INCLUDE_DIRS=/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/metis-5.1.0-6rrzripqqoszckfuh7yi3nfosysopqsh/include' '-DTPL_ENABLE_ParMETIS:BOOL=ON' '-DParMETIS_LIBRARY_DIRS=/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/parmetis-4.0.3-othzzjoecn5ggse5phajddhwkloe33w3/lib;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/metis-5.1.0-6rrzripqqoszckfuh7yi3nfosysopqsh/lib' '-DParMETIS_LIBRARY_NAMES=parmetis;metis' '-DTPL_ParMETIS_INCLUDE_DIRS=/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/parmetis-4.0.3-othzzjoecn5ggse5phajddhwkloe33w3/include;/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/metis-5.1.0-6rrzripqqoszckfuh7yi3nfosysopqsh/include' '-DTPL_ENABLE_MUMPS:BOOL=ON' '-DMUMPS_LIBRARY_DIRS=/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/mumps-5.1.1-g3i6mskheyphqju3vs2cjpjtzjz5bq56/lib' '-DMUMPS_LIBRARY_NAMES=dmumps;mumps_common;pord' '-DTPL_ENABLE_SCALAPACK:BOOL=ON' '-DSCALAPACK_LIBRARY_NAMES=scalapack' '-DSCALAPACK_LIBRARY_DIRS=/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/netlib-scalapack-2.0.2-zxvnbswlaqg7hqfcifzvm4epd4wpjic6/lib' '-DTeuchos_ENABLE_COMPLEX:BOOL=OFF' '-DKokkosTSQR_ENABLE_Complex:BOOL=OFF' '-DTPL_ENABLE_SuperLUDist:BOOL=ON' '-DSuperLUDist_LIBRARY_DIRS=/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/superlu-dist-5.3.0-26jrwc63ytki3dooo75r2pnltpi2w7x6/lib' '-DSuperLUDist_INCLUDE_DIRS=/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/superlu-dist-5.3.0-26jrwc63ytki3dooo75r2pnltpi2w7x6/include' '-DHAVE_SUPERLUDIST_LUSTRUCTINIT_2ARG:BOOL=ON' '-DTPL_ENABLE_SuperLU:BOOL=OFF' '-DTPL_ENABLE_Pnetcdf:BOOL=OFF' '-DTPL_ENABLE_Zlib:BOOL=OFF' '-DTPL_ENABLE_GGNS:BOOL=OFF' '-DTrilinos_EXTRA_LINK_FLAGS:STRING=-L/Users/davydden/spack/opt/spack/darwin-mojave-x86_64/clang-10.0.0-apple/gcc-8.2.0-345ps2sjal7bqatxvwaif2jnfp77qsau/lib/gcc/x86_64-apple-darwin18.0.0/8.2.0/../../../ -lgfortran' '-DTrilinos_ENABLE_Fortran=ON' '-DTrilinos_ENABLE_EXPLICIT_INSTANTIATION:BOOL=ON' '-DTpetra_INST_DOUBLE:BOOL=ON' '-DTpetra_INST_INT_UNSIGNED_LONG=ON' '-DTpetra_INST_INT_LONG:BOOL=ON' '-DTpetra_INST_COMPLEX_DOUBLE=ON' '-DTpetra_INST_COMPLEX_FLOAT=ON' '-DTpetra_INST_FLOAT=ON' '-DTeuchos_ENABLE_COMPLEX=ON' '-DTeuchos_ENABLE_FLOAT=ON' '-DTrilinos_ENABLE_FEI=OFF' '-DCMAKE_MACOSX_RPATH=ON' '-DCMAKE_CXX_FLAGS:STRING=-DMUMPS_5_0' '-DTrilinos_ENABLE_Pike=OFF'
```

**Comment by jhux2** [comment]
@trilinos/amesos2

**Comment by srajama1** [comment]
superlu_dist 5.3.0 support is very new. We added this as a patch recently from a user. We will take care of this. 

@jwillenbring : We need those nightly tests with superlu*.

**Comment by srajama1** [comment]
@ndellingwood

**Comment by ndellingwood** [comment]
@srajama1 This error was reported using an ancient version of Trilinos (12.12.1). We can test with the Trilinos develop branch and if the error occurs there fix it, this will be backlogged until after kokkos+kokkos-kernels integration testing is complete before I look into it.

**Comment by davydden** [comment]
> This error was reported using an ancient version of Trilinos (12.12.1)

hm, that's the latest stable release ;-) Would be good if this is fixed before `12.14.x`.

**Comment by srajama1** [comment]
@davydden The release of Trilinos has become almost redundant, other than for some very special cases, due to the master branch being used by most of our users and we the develop branch for development. Do you have any specific restrictions that will stop you from using the master branch ?

**Comment by ndellingwood** [comment]
@davydden I think that release is around 3 years old, suggested use is the master branch of Trilinos. We'll try and get this fixed by 12.14.x.

**Comment by davydden** [comment]
> Do you have any specific restrictions that will stop you from using the master branch ?

reproducibility of builds and calculations is IMO important. But I think we are getting away from the actual issue reported here.

> I think that release is around 3 years old

`29 Sep 2017` https://github.com/trilinos/Trilinos/releases  
Did you  by accident mis-read the version number?

**Comment by ndellingwood** [comment]
> Did you by accident mis-read the version number?

Yes I did, thanks for the correction.

**Comment by bangerth** [comment]
@srajama1 -- I'd venture the guess that the majority of Trilinos users outside the labs actually use the releases. Certainly that's everyone of the deal.II users, and probably also developers. (@davydden reports this from his work on deal.II.)

**Comment by srajama1** [comment]
@bangerth Thanks for the context. I see why you would want to stick with the release. We will get this fixed in 12.14.x. SuperLU-dist 5.3 was not supported with 12.12.x. With the new patches we put in it might already work with float and SuperLU-dist 5.3 (this issue). We just have to check it and then test it nightly to support it.

BTW, the master branch model is becoming popular even outside the labs. You might want to explore using the master branch as you go through the development through out the year.

**Comment by bangerth** [comment]
On 10/04/2018 10:38 AM, Siva Rajamanickam wrote:
> BTW, the master branch model is becoming popular even outside the labs. 
> You might want to explore using the master branch as you go through the 
> development through out the year.

I think it boils down whether you're a user-developer, or just a user. I 
know that I could periodically upgrade my Trilinos installation and it's 
not major additional work because I compile deal.II all the time anyway.

But all of that is not true for people who are "just" users. They just 
want a stable foundation that they compile once and then never care 
about any more. I suspect they will always want to stick with releases.

**Comment by srajama1** [comment]
@bangerth - Agreed. We want to continue the support for the user base who want to compile once and forget it. It is just balancing priorities. Our frequency of the releases have gone down from 2 to 4 an year to one an year. We will revisit these among the Trilinos leadership team and see if we can find a balance. It shouldn't be hard to release a tarball of our master branch 4 times an year. Thanks for the feedback.

**Comment by davydden** [comment]
@srajama1 @ndellingwood FYI, `trilinos-release-12-14-branch` does not build with `superlu-dist@5.4.0` either (although with a different error):
```
/Users/davydden/spack/var/spack/stage/trilinos-12.14-totwv2hz2r5ei2nwexccxz3u643ty4qz/Trilinos/packages/amesos/src/Amesos_Superludist.cpp:475:75: error: use of undeclared identifier 'LargeDiag'
     6503        if( RowPerm_ == "LargeDiag" ) PrivateSuperluData_->options_.RowPerm = LargeDiag;
     6504                                                                              ^
```
If would be great if you can fix that prior to `12.4` release.

**Comment by davydden** [comment]
@srajama1 @ndellingwood  I guess you should cherry-pick https://github.com/trilinos/Trilinos/pull/3209 PR to the 12.14 branch, I completely forgot about it.

@mhoemmen could you please cherry-pick that PR?

**Comment by jwillenbring** [comment]
@davydden I have been thinking about adding PR testing for the release branch. This would give people more confidence in a situation like this and protect the branch from obvious breakage.

I will try to do this, or ask someone else to, before COB Wednesday.

**Comment by ndellingwood** [comment]
PR #4567 is merged into Trilinos' develop branch and should address this problem.

**Comment by davydden** [comment]
Thanks, I am closing this for now and will re-open if the issue persists.

**Comment by ndellingwood** [comment]
@davydden should I also issue PR to the 12.14 branch with these changes? I saw that mentioned earlier in comments as the branch being used.

**Comment by davydden** [comment]
> should I also issue PR to the 12.14 branch with these changes?

that would be good, thanks 👍

**Comment by ndellingwood** [comment]
@davydden I put in a matching PR #4585 to the 12.14 release branch, it needs to wait on the auto-tester to complete, I pinged you on that issue so I think it will let you know once it is merged into that branch.
